### PR TITLE
Feat/coin settings facelift

### LINKED ISSUE #4727: Custom backend UI
Current draft solution with coin-chips has few weaknesses.

- Direct coin settings option slides in by hover (not applicable on touch devices, since touching has already different action than hover)

https://user-images.githubusercontent.com/26002916/149172289-5438ef6e-6af1-4449-8d11-2cc6cd541e62.mov

- Animation hides coin icon (Original idea with an increase in width may end-up with bouncing loop)
- Small grey text carrying the info about usage of Custom Backend is already taken by ETH tokens info

image

### PR DESCRIPTION
**Pull request by marekrjpolak**
Updated UI for coin selecting and settings, especially custom backends. Resolves #4727, prerequisite for #2737. Main changes:

- backends icon in navbar, with overview of all current non-default backends
- selection of enabled coins unified in onboarding and settings, with a new way of opening settings modal
  - should work a bit different on touch-only devices, so now it's possible to open coin settings on them
- coin settings modal reworked and prepared for adding other backend types (e.g. Electrum)
- IndexedDB migration to convert already saved custom backends to a new format

Demonstration: https://suite.corp.sldev.cz/suite-web/feat/coin-settings-facelift/web
Designs in Figma: https://www.figma.com/file/stayh4edU3aeLULpcvYSMu/Exploring-Settings?node-id=641%3A643

### CODE REVIEW

File: packages/suite/src/components/suite/TooltipSymbol/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Tooltip, Icon, useTheme } from '@trezor/components';
+
+const InlineTooltip = styled(Tooltip)`
+    display: inline-block;
+    margin: 0 4px;
+`;
+
+type Props = {
```
Comment (tomasklim) [inline packages/suite/src/components/suite/TooltipSymbol/index.tsx]:
```suggestion
type TooltipSymbolProps = {
```

Comment (tomasklim) [reply packages/suite/src/components/suite/TooltipSymbol/index.tsx]: Code improvements session 🤷‍♂️

File: packages/suite/src/views/settings/coins/index.tsx [hunk 2]
```diff
@@ -1,88 +1,36 @@
...
-                        ...NETWORKS.filter(testnetNetworksFilterFn)
-                            .map(n => n.symbol)
-                            .filter(unavailableNetworksFilterFn),
-                    ])
-                }
-                onDeactivateAll={() => changeNetworks(enabledMainnetNetworks)}
-                type="testnet"
-            />
-        </SettingsLayout>
+        <StyledSettingsLayout>
```
Comment (tomasklim) [inline packages/suite/src/views/settings/coins/index.tsx:22]: add titles to be consistent with other sections

File: packages/suite/src/actions/settings/constants/walletSettings.ts [hunk 3]
```diff
@@ -7,3 +7,5 @@ export const ADD_BLOCKBOOK_URL = '@wallet-settings/add-blockbook-url';
 export const REMOVE_BLOCKBOOK_URL = '@wallet-settings/remove-blockbook-url';
 export const CLEAR_TOR_BLOCKBOOK_URLS = '@wallet-settings/clear-tor-blockbook-urls';
 export const FROM_STORAGE = '@wallet-settings/from-storage';
+export const SET_BACKEND = '@backends/set-backend';
+export const REMOVE_BACKEND = '@backends/remove-backend';
```
Comment (tomasklim) [inline packages/suite/src/actions/settings/constants/walletSettings.ts]:
```suggestions
export const SET_BACKEND = '@wallet-settings/set-backend';
export const REMOVE_BACKEND = '@wallet-settings/remove-backend';
```

File: packages/suite/src/components/suite/CoinsGroup/CoinsGroupHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        {settingsMode ? (
+            <Translation id="TR_SELECT_COIN_FOR_SETTINGS" />
+        ) : (
+            <CoinsCount>
+                {total} {label}
+                {' • '}
+                {active} <Translation id="TR_ACTIVE" />
+            </CoinsCount>
+        )}
+        <SettingsWrapper onClick={toggleSettingsMode}>
```
Comment (tomasklim) [inline packages/suite/src/components/suite/CoinsGroup/CoinsGroupHeader.tsx]: Icon should be disabled when no coins selected

File: packages/suite/src/components/settings/SettingsLayout/index.tsx [hunk 5]
```diff
@@ -15,6 +15,7 @@ type Props = {
     title?: string;
     children?: React.ReactNode;
     ['data-test']?: string;
+    className?: string;
```
Comment (tomasklim) [inline packages/suite/src/components/settings/SettingsLayout/index.tsx:18]: Probably unused?

File: packages/suite/src/components/suite/Coin/index.tsx [hunk 6]
```diff
@@ -6,7 +6,7 @@ import { Translation } from '@suite-components';
 import type { ExtendedMessageDescriptor } from '@suite-types';
 import type { Network } from '@wallet-types';
 
-const SettingsWrapper = styled.div`
+const SettingsWrapper = styled.div<{ onClick: any }>`
```
Comment (tomasklim) [inline packages/suite/src/components/suite/Coin/index.tsx]: any 😭

File: packages/suite/src/components/suite/NavigationBar/components/NavigationActions/components/BackendsDropdown/index.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    const customBackends: BackendSettings[] = (
+        Object.entries(backends) as Array<
+            [Network['symbol'], NonNullable<typeof backends[Network['symbol']]>]
+        >
+    )
+        .filter(([, { tor }]) => !tor)
+        .map(([coin, settings]) => ({ coin, ...settings }));
+
+    return (
+        <Wrapper {...props}>
```
Comment (tomasklim) [inline packages/suite/src/components/suite/NavigationBar/components/NavigationActions/components/BackendsDropdown/index.tsx]: Specify marginLeft explicitly here

File: packages/suite/src/components/suite/modals/AdvancedCoinSettings/components/CustomBlockbookUrls.tsx [hunk 8]
```diff
@@ -1,159 +1,344 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/AdvancedCoinSettings/components/CustomBlockbookUrls.tsx]: This file is kinda long now

File: packages/suite/src/views/onboarding/index.tsx [hunk 9]
```diff
@@ -58,6 +66,7 @@ const Onboarding = ({ prerequisite }: InjectedModalApplicationProps) => {
 
     return (
         <LayoutComponent>
+            {useShowModal() ? modal : null /* is it ok? */}
```
Comment (tomasklim) [inline packages/suite/src/views/onboarding/index.tsx]: Backup and pin components have `Skip modal`, try to inspire there how they did it.

File: packages/suite/src/components/settings/SettingsLayout/index.tsx (hunk 5 above)
Comment (marekrjpolak) [reply packages/suite/src/components/settings/SettingsLayout/index.tsx:18]: Used here: https://github.com/trezor/trezor-suite/blob/33ebbe146e555a40a4306c83066a74df595f7883/packages/suite/src/views/settings/coins/index.tsx#L7

File: packages/suite/src/support/messages.ts [hunk 10]
```diff
@@ -5837,6 +5857,10 @@ export default defineMessages({
         id: 'TR_CUSTOM_BACKEND_BACKEND_ALREADY_ADDED',
         defaultMessage: 'The backend is already added',
     },
+    TR_OTHER_COINS_USE_DEFAULT_BACKEND: {
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts:5864]: Screenshot 2022-01-21 at 12 51 07

Better wording / two strings when I have no custom backend / no icon when I have no custom backend.

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: packages/suite/src/views/settings/coins/index.tsx (hunk 2 above)
Comment (marekrjpolak) [reply packages/suite/src/views/settings/coins/index.tsx:22]: https://github.com/trezor/trezor-suite/pull/4779/commits/cdbebe64bb56d46e7387e506b12375aab8b31f74

File: packages/suite/src/components/suite/TooltipSymbol/index.tsx (hunk 1 above)
Comment (marekrjpolak) [reply packages/suite/src/components/suite/TooltipSymbol/index.tsx]: https://github.com/trezor/trezor-suite/pull/4779/commits/0c3baa9fab227f4db586e6bd8757fd76da26ffbd

File: packages/suite/src/actions/settings/constants/walletSettings.ts (hunk 3 above)
Comment (marekrjpolak) [reply packages/suite/src/actions/settings/constants/walletSettings.ts]: https://github.com/trezor/trezor-suite/pull/4779/commits/e8b940534cd044110a2017c768bed758c16a0335

File: packages/suite/src/components/suite/CoinsGroup/CoinsGroupHeader.tsx (hunk 4 above)
Comment (marekrjpolak) [reply packages/suite/src/components/suite/CoinsGroup/CoinsGroupHeader.tsx]: https://github.com/trezor/trezor-suite/pull/4779/commits/d904cefcd47d2e22f7dd0ea1739ae06485cd4d20

File: packages/suite/src/components/suite/Coin/index.tsx (hunk 6 above)
Comment (marekrjpolak) [reply packages/suite/src/components/suite/Coin/index.tsx]: https://github.com/trezor/trezor-suite/pull/4779/commits/cbd0b063dfbaee1403268d884afa73271de7e993

File: packages/suite/src/components/suite/NavigationBar/components/NavigationActions/components/BackendsDropdown/index.tsx (hunk 7 above)
Comment (marekrjpolak) [reply packages/suite/src/components/suite/NavigationBar/components/NavigationActions/components/BackendsDropdown/index.tsx]: https://github.com/trezor/trezor-suite/pull/4779/commits/c1e8f4cb9907b1bcf4fb99acbf738604a870e1f7

File: packages/suite/src/support/messages.ts (hunk 10 above)
Comment (marekrjpolak) [reply packages/suite/src/support/messages.ts:5864]: https://github.com/trezor/trezor-suite/pull/4779/commits/241d22772929ad04a8a1552a4dcb4d4c833f039b

File: packages/suite/src/components/suite/modals/AdvancedCoinSettings/components/CustomBlockbookUrls.tsx (hunk 8 above)
Comment (marekrjpolak) [reply packages/suite/src/components/suite/modals/AdvancedCoinSettings/components/CustomBlockbookUrls.tsx]: https://github.com/trezor/trezor-suite/pull/4779/commits/c1ce85d1c5622a8bff596c4446a7f5bfc00e9f40

File: packages/suite/src/views/onboarding/index.tsx (hunk 9 above)
Comment (marekrjpolak) [reply packages/suite/src/views/onboarding/index.tsx]: The Skip modal is rendered differently, so I prefer to keep it this way. However, already present guide/modal interference in onboarding must be fixed for this modal as well as part of #4755.

**Review by tomasklim — APPROVED** [review]
LGTM

**Pull request merged**